From 1f143b3b520a12d46f08c45cd484a8327d15cd74 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Wed, 2 Mar 2016 09:26:56 -0600
Subject: [PATCH 1/8] arch/arm64: Update Axxia DTS Files

Remove inclusion of skeleton.dtsi to allow systems with
more than 4 GB of memory to be used.

Signed-off-by: John Jacques <john.jacques@intel.com>
[Original patch is from linux-yocto-4.1 axxia/base branch.]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm64/boot/dts/intel/axc67xx.dtsi |   13 +++++++++++--
 arch/arm64/boot/dts/intel/axm56xx.dtsi |   15 +++++++++++++--
 2 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/axc67xx.dtsi b/arch/arm64/boot/dts/intel/axc67xx.dtsi
index 6ddff94..d3d1115 100644
--- a/arch/arm64/boot/dts/intel/axc67xx.dtsi
+++ b/arch/arm64/boot/dts/intel/axc67xx.dtsi
@@ -12,11 +12,20 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/clock/lsi,axm5516-clks.h>
 
-#include "skeleton.dtsi"
-
 / {
+	#address-cells = <2>;
+	#size-cells = <1>;
+
 	interrupt-parent = <&gic>;
 
+	chosen {
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0 0 0>;
+	};
+
 	aliases {
 		serial0	  = &serial0;
 		serial1   = &serial1;
diff --git a/arch/arm64/boot/dts/intel/axm56xx.dtsi b/arch/arm64/boot/dts/intel/axm56xx.dtsi
index 1c0b372..de96ecb 100644
--- a/arch/arm64/boot/dts/intel/axm56xx.dtsi
+++ b/arch/arm64/boot/dts/intel/axm56xx.dtsi
@@ -12,11 +12,22 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/clock/lsi,axm5516-clks.h>
 
-#include "skeleton.dtsi"
-
 / {
+	#address-cells = <2>;
+	#size-cells = <1>;
+
 	interrupt-parent = <&gic>;
 
+	chosen {
+	};
+
+	memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		device_type = "memory";
+		reg = <0 0 0 0>;
+	};
+
 	aliases {
 		serial0	  = &serial0;
 		serial1   = &serial1;
-- 
1.7.5.4

